lib/gc/Transforms/Pipeline.cpp

```diff
@@ -95,6 +95,7 @@ void populateBufferizationPasses(mlir::OpPassManager &pm) {
   options.setFunctionBoundaryTypeConversion(
       bufferization::LayoutMapOption::IdentityLayoutMap);
   pm.addPass(bufferization::createOneShotBufferizePass(options));
+  pm.addPass(createCanonicalizerPass());
```

Comment:
The `canonicalizerPass` could canonicalize the `scf.for` in memref semantic so that `BufferResultsToOutParams` could remove the last `memref.copy`